react/tic-tac-toe/src/app/screens/Game/components/Board/index.js

```diff
@@ -1,37 +1,48 @@
-import React, { Component } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
 
 import Square from '../Square';
 
 import styles from './styles.module.scss';
 
-class Board extends Component {
-  renderSquare() {
-    return <Square />;
-  }
-
-  render() {
-    const status = 'Next player: X';
-    return (
-      <div>
-        <div className={styles.status}>{status}</div>
-        <div className={styles.boardRow}>
-          {this.renderSquare(0)}
-          {this.renderSquare(1)}
-          {this.renderSquare(2)}
-        </div>
-        <div className={styles.boardRow}>
-          {this.renderSquare(3)}
-          {this.renderSquare(4)}
-          {this.renderSquare(5)}
-        </div>
-        <div className={styles.boardRow}>
-          {this.renderSquare(6)}
-          {this.renderSquare(7)}
-          {this.renderSquare(8)}
-        </div>
+function Board ({ status, squares, handleClick }) {
+  return (
+    <div>
+      <div className={styles.status}>{status}</div>
+      <div className={styles.boardRow}>
+        <Square value={squares[0]} index={0} onClick={handleClick} />
+        <Square value={squares[1]} index={1} onClick={handleClick} />
+        <Square value={squares[2]} index={2} onClick={handleClick} />
       </div>
-    );
-  }
+      <div className={styles.boardRow}>
+        <Square value={squares[3]} index={3} onClick={handleClick} />
+        <Square value={squares[4]} index={4} onClick={handleClick} />
+        <Square value={squares[5]} index={5} onClick={handleClick} />
+      </div>
+      <div className={styles.boardRow}>
+        <Square value={squares[6]} index={6} onClick={handleClick} />
+        <Square value={squares[7]} index={7} onClick={handleClick} />
+        <Square value={squares[8]} index={8} onClick={handleClick} />
+      </div>
+    </div>
+  );
 }
 
+Board.propTypes = {
+  handleClick: PropTypes.func.isRequired,
+  squares: PropTypes.arrayOf(PropTypes.string).isRequired,
+  status: PropTypes.string.isRequired
+ };
+
 export default Board;
+
+export function calculateWinner(squares) {
+  const lines = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]];
```

Comment:
Move this function to utils.js,  where we always define the auxiliary functions, and then import it on this component.